WalletWasabi.Tests/UnitTests/WabiSabi/Integration/WabiSabiHttpApiIntegrationTests.cs

```diff
@@ -436,17 +444,35 @@ public async Task MultiClientsCoinJoinTestAsync(int seed)
 					services.AddScoped<WabiSabiConfig>(s => new WabiSabiConfig(Path.GetTempFileName())
 					{
 						MaxRegistrableAmount = Money.Coins(500m),
-						MaxInputCountByRound = ExpectedInputNumber,
-						StandardInputRegistrationTimeout = TimeSpan.FromSeconds(10 * ExpectedInputNumber),
-						ConnectionConfirmationTimeout = TimeSpan.FromSeconds(20 * ExpectedInputNumber),
-						OutputRegistrationTimeout = TimeSpan.FromSeconds(20 * ExpectedInputNumber),
-						TransactionSigningTimeout = TimeSpan.FromSeconds(20 * ExpectedInputNumber),
+						MaxInputCountByRound = (int)(ExpectedInputNumber / (1 + 10 * (faultInjectorMonkeyAggressiveness + delayInjectorMonkeyAggressiveness)) ),
+						StandardInputRegistrationTimeout = TimeSpan.FromSeconds(5 * ExpectedInputNumber),
+						BlameInputRegistrationTimeout = TimeSpan.FromSeconds(2 * ExpectedInputNumber),
+						ConnectionConfirmationTimeout = TimeSpan.FromSeconds(2 * ExpectedInputNumber),
+						OutputRegistrationTimeout = TimeSpan.FromSeconds(3 * ExpectedInputNumber),
+						TransactionSigningTimeout = TimeSpan.FromSeconds(2 * ExpectedInputNumber),
 						MaxSuggestedAmountBase = Money.Satoshis(ProtocolConstants.MaxAmountPerAlice)
 					});
 				}));
 
 			using PersonCircuit personCircuit = new();
-			IHttpClient httpClientWrapper = new HttpClientWrapper(app.CreateClient());
+			IHttpClient httpClientWrapper = new HttpClientWrapperWithMonkeys(
```

Comment:
I did it in that way because some monkeys can be specialized, for example they could break specific phases and similar things. I prefer to let is as it is. In fact the first design had only one monkey.